### ISSUE TITLE
Case class extraction

### ISSUE THREAD

**Issue by lorenz-nlp** [akka/akka-http#3000]
### Versions:
Scala version: 2.11.12
Akka-Http version: 10.1.11

### Description:
A path element shall be converted into a case class instance. The case class has an explicit companion object defined.

```
case class ColorName(name: String)
object ColorName { 
  // ...
}
```
As the case class does have an explicit companion object, following code needs to be used according to the documentation (https://doc.akka.io/docs/akka-http/current/routing-dsl/case-class-extraction.html) for converting into a case class instance:

```
val route = path("color" / Segment).as(ColorName.apply) { colorName =>
	complete(colorName.name)
}
```
However, this results into a compile error at "ColorName.apply".

### Possible solution:
If an underscore if added behind apply, no compile error appears anymore:
```
val route = path("color" / Segment).as(ColorName.apply _) { colorName =>
	complete(colorName.name)
}
```
Therefore, I suggest to verify and update the documentation accordingly.

**Comment by jrudolph** [comment]
Thanks for the report. Would you like to open a PR for that?

**Comment by ennru** [comment]
Fixed with #3092
